pkg/skaffold/build/docker/types.go

```diff
@@ -31,15 +34,21 @@ type Builder struct {
 	mode               config.RunMode
 	insecureRegistries map[string]bool
 	artifacts          ArtifactResolver
+	deps               DependencyResolver
 }
 
 // ArtifactResolver provides an interface to resolve built artifact tags by image name.
 type ArtifactResolver interface {
 	GetImageTag(imageName string) (string, bool)
 }
 
+// ArtifactResolver provides an interface to resolve built artifact tags by image name.
```

Comment:
This should likely be `DependenciesResolver` though. 

```suggestion
// DependencyResolver provides an interface to resolve built artifact tags by image name.
```

Why can't the code use `pkg/skaffold/build.DependencyResolver` though?

Reply:
package `build` imports `docker` so it can't import the interface from there. Also I can't move the interface definition entirely to `docker` package because then all the other builders will need to take a dependency on `docker`.